### ISSUE TITLE
Perform evaluation on validation every n epochs

### ISSUE THREAD

**Issue by diningphil** [diningphil/PyDGN#8]
### Feature description

One could specify the number of epochs after which to perform evaluation on validation and on test (depending if this is the final training run and we want to log test performances).

[Optional] Also, it would make sense not to re-evaluate on the entire validation set, if this is very huge, and instead specify a "mini-batch" or something similar.

### Ideas on how to do it

This will affect how the EarlyStopper computes its own scores, so please keep this in mind.
